src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -0,0 +1,218 @@
+                }
+
+                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestPath))
+                {
+                    AddToList(list, nameof(request.PathBase), request.PathBase);
+                    AddToList(list, nameof(request.Path), request.Path);
+                }
+
+                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestQuery))
+                {
+                    AddToList(list, nameof(request.QueryString), request.QueryString.Value);
+                }
+
+                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestHeaders))
+                {
+                    FilterHeaders(list, request.Headers, options.RequestHeaders);
+                }
+
+                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestBody)
+                    && MediaTypeHelpers.TryGetEncodingForMediaType(request.ContentType,
+                        options.SupportedMediaTypes,
+                        out var encoding))
+                {
+                    originalBody = request.Body;
+                    requestBufferingStream = new RequestBufferingStream(
+                        request.Body,
+                        options.RequestBodyLogLimit,
+                        _logger,
+                        encoding);
+                    request.Body = requestBufferingStream;
+                }
+
+                // TODO add and remove things from log.
+                var httpRequestLog = new HttpRequestLog(list);
+
+                _logger.Log(LogLevel.Information,
+                     eventId: LoggerEventIds.RequestLog,
+                     state: httpRequestLog,
+                     exception: null,
+                     formatter: HttpRequestLog.Callback);
+            }
+
+            ResponseBufferingStream? responseBufferingStream = null;
+            IHttpResponseBodyFeature? originalBodyFeature = null;
+
+            try
+            {
+                if ((HttpLoggingFields.Response & options.LoggingFields) == HttpLoggingFields.None)
+                {
+                    // Short circuit and don't replace response body.
+                    await _next(context).ConfigureAwait(false);
+                    return;
+                }
+
+                var response = context.Response;
+
+                if (options.LoggingFields.HasFlag(HttpLoggingFields.ResponseBody))
+                {
+                    originalBodyFeature = context.Features.Get<IHttpResponseBodyFeature>()!;
+                    // TODO pool these.
+                    responseBufferingStream = new ResponseBufferingStream(originalBodyFeature,
+                        options.ResponseBodyLogLimit,
+                        _logger,
+                        context,
+                        options.SupportedMediaTypes);
+                    response.Body = responseBufferingStream;
+                }
+
+                await _next(context).ConfigureAwait(false);
```

Comment:
How come you've added ConfigureAwait(false)? Won't this only ever run inside ASP.NET Core and so will never have a sync context.

Reply:
Yea remove it.